### PR TITLE
feat: implement new helix schedule api

### PR DESCRIPTION
**Pull request by iProdigy**
### Prerequisites for Code Changes
* [x] This pull request follows the code style of the project
* [ ] I have tested this feature

### Changes Proposed
* Add `TwitchHelix#getChannelStreamSchedule`
* Add `TwitchHelix#getChannelInternetCalendar`
* Add `TwitchHelix#updateChannelStreamSchedule`
* Add `TwitchHelix#createStreamScheduleSegment`
* Add `TwitchHelix#updateStreamScheduleSegment`
* Add `TwitchHelix#deleteStreamScheduleSegment`

### Additional Information
https://dev.twitch.tv/docs/change-log

### CODE REVIEW

File: rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java [hunk 1]
```diff
@@ -990,6 +990,113 @@
...
+    );
+
+    /**
+     * Gets all scheduled broadcasts from a channel’s stream schedule as an iCalendar.
+     *
+     * @param broadcasterId User ID of the broadcaster who owns the channel streaming schedule.
+     * @return iCalendar data is returned according to RFC5545. The expected MIME type is text/calendar.
+     */
+    @RequestLine("GET /schedule/icalendar?broadcaster_id={broadcaster_id}")
+    HystrixCommand<Response> getChannelInternetCalendar(
```
Comment (iProdigy) [inline rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java:1024]: Is there a better return type possible here?

File: rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java [hunk 2]
```diff
@@ -990,6 +990,113 @@
...
+    /**
+     * Gets all scheduled broadcasts or specific scheduled broadcasts from a channel’s stream schedule.
+     *
+     * @param authToken     User OAuth Token or App Access Token.
+     * @param broadcasterId User ID of the broadcaster who owns the channel streaming schedule.
+     * @param ids           The ID of the stream segment to return.Maximum: 100.
+     * @param startTime     A timestamp in RFC3339 format to start returning stream segments from. If not specified, the current date and time is used.
+     * @param utcOffset     A timezone offset for the requester specified in minutes. For example, a timezone that is +4 hours from GMT would be “240.” If not specified, “0” is used for GMT.
+     * @param after         Cursor for forward pagination: tells the server where to start fetching the next set of results in a multi-page response. The cursor value specified here is from the pagination response field of a prior query.
+     * @param limit         Maximum number of stream segments to return.Maximum: 25. Default: 20.
```
Comment (iProdigy) [inline rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java:1002]:
```suggestion
     * @param ids           The ID of the stream segment to return. Maximum: 100.
     * @param startTime     A timestamp in RFC3339 format to start returning stream segments from. If not specified, the current date and time is used.
     * @param utcOffset     A timezone offset for the requester specified in minutes. For example, a timezone that is +4 hours from GMT would be “240.” If not specified, “0” is used for GMT.
     * @param after         Cursor for forward pagination: tells the server where to start fetching the next set of results in a multi-page response. The cursor value specified here is from the pagination response field of a prior query.
     * @param limit         Maximum number of stream segments to return. Maximum: 25. Default: 20.
```

File: rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java (hunk 1 above)
Comment (stachu540) [reply rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java:1024]: Possible usage with: https://github.com/ical4j/ical4j

Comment (iProdigy) [reply rest-helix/src/main/java/com/github/twitch4j/helix/TwitchHelix.java:1024]: Yeah but I also don't want to add a dependency for just that endpoint - though, I probably should add a default method that converts the Response to a String

**Review by PhilippHeuer — APPROVED** [review]

**Pull request merged**